src/js_parser.zig

```diff
@@ -18015,8 +18016,12 @@ fn NewParser_(
 
                     data.value = p.visitExpr(data.value);
 
+                    if (should_trim_primitive and data.value.isPrimitiveLiteral()) {
+                        return;
+                    }
+
                     // simplify unused
-                    data.value = SideEffects.simpifyUnusedExpr(p, data.value) orelse data.value.toEmpty();
+                    data.value = SideEffects.simpifyUnusedExpr(p, data.value) orelse return;
```

Comment:
we should always simplify unused im pretty sure

Reply:
we append it if it can be simplified, or we return and don't append the stmt. this was actually the code that replaced empty blocks with `e_missing` and made empty if/while/for statements look like this after building:
```js
if (r)
    ;
else if (r)
    ;
```
this is fine, but doesn't work when minifying syntax and whitespace later on

Reply:
oh got it cool